### DISCUSSION TITLE
With keepPreviousData on, the status doesn't change to loading

### DISCUSSION THREAD

**Discussion by hmbrg** [TanStack/query#1238] [Ideas] [answered]
I would be great to have an option that allows us to keepPreviousData but also have the status have the `loading` state when fetching data.

This would be useful because it can be utilized to show a loading state when loading takes longer than a specified amount of time. I there any way to achieve this with the current API?

**Comment by TkDodo** [comment, marked as answer]
The `loading` state signalizes a "hard" loading - which means a state where you have no data yet and also no error. You want to always show a loading spinner / indicator here. 

This is the state we want to avoid with `keepPreviousData`, so this is intentional. What you are looking for is probably the `isFetching` flag that is also returned. This flag is true whenever a request is going out, no matter why. So it will also be true for background refetches, like when you focus the window or when you are fetching with previous data kept like in your example.

**Comment by BooDeer** [comment]
You can achieve the desired behavior using a combination of `isPreviousData` and `isLoading`, you can check this StackOverflow post [https://stackoverflow.com/questions/68570071/how-to-get-the-isloading-state-while-keeppreviousdata-is-on-in-react-query](url)
